File: src/main/java/com/cegeka/employeeplanning/data/Mitarbeiter.java
package com.cegeka.employeeplanning.data;

import com.cegeka.employeeplanning.data.enums.Enums;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
public class Mitarbeiter {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    //Todo - Es ist mir nicht gelungen die Enum-Werte numerisch zu definieren und korrekt zu initialisieren.
    //Statt dessen habe ich die Enum-Werte als Notlösung als Strings definiert.
    @Enumerated(EnumType.STRING)
    private Enums.MitarbeiterStatus mitarbeiterStatus;
    private double stundensatzEK;
    private String name;
    private String vorname;
    @Enumerated(EnumType.STRING)
    private Enums.MitarbeiterUnit mitarbeiterUnit;
}

File: angularclient/src/app/mitarbeiter.ts
import {MitarbeiterStatus} from "./mitarbeiter-status.enum";
import {MitarbeiterUnit} from "./mitarbeiter-unit.enum";

export class Mitarbeiter {
  id: string;
  mitarbeiterStatus: MitarbeiterStatus;
  stundensatzEK: string;
  name: string;
  vorname: string;
  mitarbeiterUnit: MitarbeiterUnit;
}
